### PR TITLE
Refactor eatn park to StructuredDataSpider

### PR DESCRIPTION
**Pull request by CloCkWeRX**

### CODE REVIEW

File: locations/spiders/eatn_park.py [hunk 1]
```diff
@@ -1,40 +1,25 @@
...
+class EatnParkSpider(SitemapSpider, StructuredDataSpider):
     name = "eatn_park"
     item_attributes = {"brand": "Eat'n Park", "brand_wikidata": "Q5331211"}
     sitemap_urls = ["https://locations.eatnpark.com/robots.txt"]
-    sitemap_rules = [(r"/restaurants-", "parse")]
-
-    def parse(self, response):
-        ldjson = response.xpath('//script[@type="application/ld+json"]/text()')
-        [data] = json.loads(ldjson.get())
+    sitemap_rules = [(r"/restaurants-", "parse_sd")]
```
Comment (Cj-Malone) [inline locations/spiders/eatn_park.py:15]:
```suggestion
    sitemap_rules = [("/restaurants-", "parse_sd")]
```

File: locations/spiders/eatn_park.py [hunk 2]
```diff
@@ -1,40 +1,25 @@
...
 
+    def post_process_item(self, item, response, ld_data):
         opening_hours = OpeningHours()
-        for m in HOURS_RE.finditer(data["openingHours"]):
+        for m in HOURS_RE.finditer(ld_data["openingHours"]):
             g = m.groupdict()
             opening_hours.add_range(g["day"], g["open_time"], g["close_time"])
```
Comment (Cj-Malone) [inline locations/spiders/eatn_park.py:21]: Normal format? If so:
```suggestion
```
And `HOURS_RE`

File: locations/spiders/eatn_park.py [hunk 3]
```diff
@@ -1,40 +1,25 @@
...
-            "street_address": data["address"]["streetAddress"],
-            "city": data["address"]["addressLocality"],
-            "state": data["address"]["addressRegion"],
-            "postcode": data["address"]["postalCode"],
-            "opening_hours": opening_hours.as_opening_hours(),
-            "phone": data["address"]["telephone"],
-        }
-        yield Feature(**properties)
+        item["opening_hours"] = opening_hours
+        item["name"] = (response.css("span.location-name::text").get(),)
```
Comment (Cj-Malone) [inline locations/spiders/eatn_park.py:23]: Probably not a set

File: locations/spiders/eatn_park.py (hunk 2 above)
Comment (CloCkWeRX) [reply locations/spiders/eatn_park.py:21]: Lets give it a go and see

File: locations/spiders/eatn_park.py [hunk 4]
```diff
@@ -1,40 +1,19 @@
...
-            "street_address": data["address"]["streetAddress"],
-            "city": data["address"]["addressLocality"],
-            "state": data["address"]["addressRegion"],
-            "postcode": data["address"]["postalCode"],
-            "opening_hours": opening_hours.as_opening_hours(),
-            "phone": data["address"]["telephone"],
-        }
-        yield Feature(**properties)
+    def post_process_item(self, item, response, ld_data):
+        item["name"] = response.css("span.location-name::text").get()
```
Comment (Cj-Malone) [inline locations/spiders/eatn_park.py:17]:
```suggestion
        item["branch"] = response.css("span.location-name::text").get()
```

**Pull request merged**